wallet/db.c

```diff
@@ -684,6 +684,24 @@ static struct db *db_open(const tal_t *ctx, char *filename)
 	return db;
 }
 
+/**
+ * Tentatively execute a query, but allow failures.
+ *
+ * Some databases like postgres will terminate the DB transaction if there is
+ * an error during the execution of a query, e.g., trying to access a table
+ * that doesn't exist yet, so we need to terminate and restart the DB
+ * transaction.
+ */
+static bool db_query_prepared_may_fail(struct db_stmt *stmt)
+{
+	bool ok = db_query_prepared(stmt);
+	if (!ok) {
+		db_commit_transaction(stmt->db);
+		db_begin_transaction(stmt->db);
+	}
```

Comment:
This function is horrifying!  It's safe only as used here, since there's nothing else in the transaction.  Please just put this into db_get_version itself so nobody else thinks they can use it...

Reply:
Inlined :+1: